setup.py

```diff
@@ -49,6 +79,15 @@
     )
     raise RuntimeError(msg)
 
+version_hex = matches.get("HEX", "MISSING")
+expected_version_hex = build_expected_version_hex(matches)
```

Comment:
At this point, already have the `version`, can't you just compute the expected version hex from that?

Reply:
IIUC that's what I'm doing. The subproblem that makes the code long is handling `patch_level_serial`, e.g. `"3.dev1"`. And then we also need the conversion from base-10 to base-16 (hex), which doesn't happen anywhere else.

The only alternative I see is to do it the other way around, building expected_major, expected_minor, expected_patch from the hex value, but I think it'll be a very similar amount of code.

In any case, it's done "good enough for our purposes" I feel, and not that much code :-)

Reply:
I might take a stab at refactoring it eventually, but good for now. :)

Reply:
> I might take a stab at refactoring it eventually, but good for now. :)

Sounds good, as long as it doesn't look like write-only Perl at the end ;-)